Handle specific archive and book versions

Dantemss opened the pull request:
For: https://github.com/openstax/unified/issues/2610

There are a couple of failing tests in orn-locator/src/lookup.ts because of missing fetch, but that file seems to run in-browser...

Dantemss committed:
Handle specific archive and book versions

TomWoodward commented on packages/orn-locator/src/ornPatterns.ts:
```diff
@@ -20,26 +20,25 @@ export const patterns = {
     name: 'Books',
-    pattern: 'https\\://openstax.org/orn/book/:id',
-    resolve: ({id}: {id: string}) => import('./resolvers/books').then(mod => mod.bookDetail(id)),
+    pattern: 'https\\://openstax.org/orn/{archive/:archivePath/}?book/:id{@:version}?',
```
in order to keep the overall orn format more consistent, can you shove both formats on the end of the resource id?

rex uses the version format `@:book_contentVersion\::book_archiveVersion`

Dantemss replied:
I think this works as long as we force the book content and archive versions to be specified together, otherwise the `:` is ambiguous with the other `:` in the pattern

Dantemss replied:
Changed to this format, but you can only specify the archive version if you also specify the book version.

TomWoodward reviewed (changes requested):
most changes look fine, made a small request about the versioning format

Dantemss committed:
Changed the archive version pattern
Renamed some params
Fixed orn-locator tests

Dantemss committed:
Handle calling getArchiveInfo repeatedly (for commonBook)
Removed console.error

Dantemss committed:
commonBook no longer takes output of getArchiveInfo
Removed path check from getArchiveInfo
Memoized getArchiveInfo
Preserve book content/archive version in mapTree

Dantemss committed:
Pass book content and archive versions to more instances of mapTree()

Dantemss committed:
syncPageNodeData also needs the versions

Dantemss commented:
I believe this is ready for another review. I made sure it preserves the version in all the returned ORN urls I could see as well.

TomWoodward commented:
Thinking about it.... I'm not sure about preserving the versions in the returned orns... There are probably times when we want that to happen especially with the nested lookups within the service, but when we get to tagging the content we want unversioned orns

Maybe we need to add a new parameter for the versioned orn and use that for the nested lookups internally?

Dantemss commented:
The lookups seem independent of the returned ORNs so we could just undo the changes to the returned ORNs to remove the versions. But this means you may be looking at an old version and follow an ORN and accidentally end on the latest version?

Dantemss committed:
Split orn and versionedOrn

Dantemss commented:
Does this look good? I split orn and versionedOrn, but versionedOrn is only versioned when you access the orn by specifying the actual versions.
